platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java

```diff
@@ -81,17 +105,117 @@ public class InOrderLinker {
     /**
      * Constructor
      *
+     * @param platformContext    the platform context
+     * @param time               a source of time
      * @param intakeEventCounter keeps track of the number of events in the intake pipeline from each peer
      */
-    public InOrderLinker(@NonNull final IntakeEventCounter intakeEventCounter) {
+    public InOrderLinker(
+            @NonNull final PlatformContext platformContext,
```

Comment:
If we just need the metrics, can we pass is just the metrics? I am not a big fan of passing the context everywhere, it makes the dependencies less obvious and complicates unit tests

Reply:
Based on Kelly's comment in slack, leaving this as the entire context